tt_metal/common/base_types.hpp

```diff
@@ -38,3 +38,10 @@ struct std::hash<MathFidelity>
         return static_cast<std::size_t>(obj);
     }
 };
+
+enum class PreserveFP32Target : uint8_t
```

Comment:
@yan-zaretskiy @rtawfik01 could you review this impl. with @rdjogoTT ... I synced with him and I'd like your input on these changes.

Preserving fp32 implies unpacking to dst `PreserveFP32Target::SRCA_B` is just not legal at all so it was confusing to represent that as an enum type. I think we want to try to solve the problem by having a check to say preserving fp32 is not legal programming if we're unpacking to src reg

Reply:
Hi @cfjchu, I agree and I synced with Radomir to rename the struct to something that reflects `UnpacktoDest` option, and with the rename it is more evident that SrcA/B are not options and will be removed.